gapic/src/main/com/google/gapid/perfetto/views/TraceMetadataDialog.java

```diff
@@ -0,0 +1,120 @@
+import com.google.gapid.rpc.RpcException;
+import com.google.gapid.rpc.UiCallback;
+import com.google.gapid.util.Messages;
+import com.google.gapid.widgets.DialogBase;
+import com.google.gapid.widgets.Theme;
+import com.google.gapid.widgets.Widgets;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.viewers.LabelProvider;
+import org.eclipse.jface.viewers.TableViewer;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swt.widgets.TableColumn;
+
+public class TraceMetadataDialog {
+
+  private static final Logger LOG = Logger.getLogger(TraceMetadataDialog.class.getName());
+  private static final String QUERY_STR =
+      "SELECT name AS 'Name', CAST(COALESCE(int_value, str_value, 'NULL') as TEXT) as 'Value' FROM"
+          + " metadata;";
+
+  public static void showMetadata(Shell shell, Models models, Theme theme) {
+    models.analytics.postInteraction(View.Help, ClientAction.Show);
+    new DialogBase(shell, theme) {
+      @Override
+      public String getTitle() {
+        return Messages.TRACE_METADATA_VIEW_TITLE;
+      }
+
+      @Override
+      protected Control createDialogArea(Composite parent) {
+        Composite area = (Composite) super.createDialogArea(parent);
+        TableViewer table = createTableViewer(area, SWT.BORDER | SWT.H_SCROLL | SWT.V_SCROLL);
+        table.getTable().setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+        table.setContentProvider(new ResultContentProvider());
+        table.setLabelProvider(new LabelProvider());
+        Rpc.listen(
+            models.perfetto.query(QUERY_STR),
+            new UiCallback<Perfetto.QueryResult, Perfetto.QueryResult>(area, LOG) {
+              @Override
+              protected Perfetto.QueryResult onRpcThread(Result<Perfetto.QueryResult> result)
+                  throws ExecutionException {
+                try {
+                  return result.get();
+                } catch (RpcException e) {
+                  LOG.log(Level.WARNING, "System Profile Query failure", e);
+                  return Perfetto.QueryResult.newBuilder().setError(e.toString()).build();
+                }
+              }
+
+              @Override
+              protected void onUiThread(Perfetto.QueryResult result) {
+                table.setInput(null);
+                for (TableColumn col : table.getTable().getColumns()) {
+                  col.dispose();
+                }
+
+                if (!result.getError().isEmpty()) {
+                  Widgets.createTableColumn(table, "Error", $ -> result.getError());
+                } else if (result.getNumRecords() == 0) {
+                  Widgets.createTableColumn(table, "Result", $ -> "Query returned no rows.");
```

Comment:
Maybe these informing message should be informed in a Label rather than in the table?